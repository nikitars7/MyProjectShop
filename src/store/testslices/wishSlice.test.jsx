import wishListReducer, {
  addToWishList,
  removeFromWishList,
  plusItem,
  minusItem,
} from "../slices/wishSlice";

describe("wishSlice", () => {
  test("should return default state", () => {
    const result = wishListReducer(undefined, { type: "" });
    expect(result).toEqual({ wishItems: [] });
    expect(result).toMatchSnapshot();
  });
  test("should add a new item to the state", () => {
    const action = {
      type: addToWishList.type,
      payload: {
        id: 1,
        imageUrl: "image",
        name: "added",
        price: 100,
        count: 1,
        size: "L",
      },
    };
    const result = wishListReducer({ wishItems: [] }, action);
    const { wishItems } = result;
    expect(wishItems[0].name).toBe("added");
    expect(result).toMatchSnapshot();
  });
  test("should remove the item from the state", () => {
    const wishItem = {
      id: 1,
      imageUrl: "image",
      name: "added",
      price: 100,
      count: 1,
      size: "L",
    };
    const action = {
      type: removeFromWishList.type,
      payload:1,
    };
    const result = wishListReducer({ wishItems: [wishItem] }, action);
    expect(result).toEqual({ wishItems: [] });
    expect(result).toMatchSnapshot();
  });
  test("should add count to the item", () => {
    const wishItem = {
      id: 1,
      imageUrl: "image",
      name: "added",
      price: 100,
      count: 1,
      size: "L",
    };
    const action = {
      type: plusItem.type,
      payload:{
        id: 1,
        imageUrl: "image",
        name: "added",
        price: 100,
        count: 1,
        size: "L",
      },
    };
    const result = wishListReducer({ wishItems: [wishItem] }, action);
    const { wishItems } = result;
    expect(wishItems[0].count).toBe(2);
    expect(result).toMatchSnapshot();
  });
  test("should minus count of the item", () => {
    const wishItem = {
      id: 1,
      imageUrl: "image",
      name: "added",
      price: 100,
      count: 2,
      size: "L",
    };
    const action = {
      type: minusItem.type,
      payload:{
        id: 1,
        imageUrl: "image",
        name: "added",
        price: 100,
        count: 1,
        size: "L",
      },
    };
    const result = wishListReducer({ wishItems: [wishItem] }, action);
    const { wishItems } = result;
    expect(wishItems[0].count).toBe(1);
    expect(result).toMatchSnapshot();
  });
});
