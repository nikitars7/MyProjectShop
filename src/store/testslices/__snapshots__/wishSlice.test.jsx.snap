// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wishSlice should add a new item to the state 1`] = `
Object {
  "wishItems": Array [
    Object {
      "count": 1,
      "id": 1,
      "imageUrl": "image",
      "name": "added",
      "price": 100,
      "size": "L",
    },
  ],
}
`;

exports[`wishSlice should add count to the item 1`] = `
Object {
  "wishItems": Array [
    Object {
      "count": 2,
      "id": 1,
      "imageUrl": "image",
      "name": "added",
      "price": 100,
      "size": "L",
    },
  ],
}
`;

exports[`wishSlice should minus count of the item 1`] = `
Object {
  "wishItems": Array [
    Object {
      "count": 1,
      "id": 1,
      "imageUrl": "image",
      "name": "added",
      "price": 100,
      "size": "L",
    },
  ],
}
`;

exports[`wishSlice should remove the item from the state 1`] = `
Object {
  "wishItems": Array [],
}
`;

exports[`wishSlice should return default state 1`] = `
Object {
  "wishItems": Array [],
}
`;
